src/MBC_TransactionalEmail_Consumer.php

```diff
@@ -223,68 +229,112 @@ protected function process() {
+            }
+            else {
+              $templateName = 'mb-' . str_replace('_', '-', $message['activity']) . '-XG';
+            }
+            break;
+
+          case 'en-global':
+            $templateName = 'mb-' . str_replace('_', '-', $message['activity']) . '-XG';
+            break;
+
+          case 'es-mx':
+            $templateName = 'mb-' . str_replace('_', '-', $message['activity']) . '-MX';
+            break;
+
+          case 'pt-br':
+            $templateName = 'mb-' . str_replace('_', '-', $message['activity']) . '-BR';
+            break;
+
+        }
+        break;
+
+      case "vote":
+
+        if (strtoupper($message['user_country']) == 'US') {
+          $templateName = 'mb-cgg-vote-US';
+        }
+        else {
+          $templateName = 'mb-cgg-vote-XG';
+        }
+        break;
+
+       default:
+         $templateName = FALSE;
+
     }
-    elseif (isset($message['email-template'])) {
-      $template = $message['email-template'];
+
+    if ($templateName == FALSE) {
+      $statName = 'mbc-transactional-email: Invalid Template';
+      $this->statHat->ezCount($statName, 1);
     }
     else {
-      throw new Exception('Template not defined : ' . print_r($message, TRUE));
-      $template = FALSE;
-    }
-
-    $templateBits = explode('-', $template);
-    $countryCode = $templateBits[count($templateBits) - 1];
-    // Don't apply country code logic to transactional requests that define source
-    // A source value suggests user import or other app that doesn't want the
-    // default template name generation.
-    if (!(isset($message['source']))) {
-
-      // mb-campaign-signup-KR
-      if ($message['user_country'] == 'US') {
-        $templateName = $template;
-      }
-      elseif ($this->mbToolbox->isDSAffiliate($countryCode)) {
-        $templateName = $template;
-      }
-      else {
-        $templateName = 'mb-' . str_replace('_', '-', $message['activity']) . '-GL';
-      }
-
-      echo '- activity: ' . $message['activity'], PHP_EOL;
       $statName = 'mbc-transactional-email: activity: ' . $message['activity'];
       $this->statHat->ezCount($statName, 1);
-      echo '- countryCode: ' . $countryCode, PHP_EOL;
       $statName = 'mbc-transactional-email: country: ' . $countryCode;
       $this->statHat->ezCount($statName, 1);
-
+      $statName = 'mbc-transactional-email: template: ' . $templateName;
+      $this->statHat->ezCount($statName, 1);
     }
-    elseif ($message['application_id'] == 'CGG') {
-      $templateCountries = ['US', 'MX', 'BR'];
-      if (!(in_array($countryCode, $templateCountries))) {
-        $templateName = 'mb-cgg2015-vote-GL';
+
+    return $templateName;
+  }
+
+  /*
+   * logConsumption(): Log the status of processing a specific message element.
+   *
+   * @param string $targetName
+   */
+  protected function logConsumption($targetName = NULL) {
+
+    if ($targetName != NULL) {
```

Comment:
You better check if it is NULL and return from it so the actual logic is not indented in if.